### ISSUE TITLE
Error: failed to execute script main

### ISSUE THREAD

**Issue by feiercrack** [NathanVaughn/msfs-mod-manager#32]
when  trying to launch for the 1st time, an error appears. before it found the correct installation folder.. but then nothing more..

"failed to execute script main "

**Comment by NathanVaughn** [comment]
Sorry for the delay, could you post the end of your `debug.log` file, so I can figure out what's going on? This is found at `%APPDATA%\MSFS Mod Manager\debug.log`.

**Comment by feiercrack** [comment]
beginning:
UnicodeDecodeError: 'charmap' codec can't decode byte 0x81 in position 951: character maps to <undefined>
2020-10-03 01:36:57.011 | INFO     | __main__:<module>:28 - -----------------------
2020-10-03 01:36:57.011 | INFO     | __main__:<module>:29 - Launching application
2020-10-03 01:36:57.011 | DEBUG    | lib.version:get_version:32 - Attemping to determine current application version
2020-10-03 01:36:57.011 | DEBUG    | lib.version:get_version:34 - Parsing D:\D-Games\MSFS Mod Manager\base.json
2020-10-03 01:36:57.013 | DEBUG    | lib.version:get_version:38 - Version found: v0.4.1
2020-10-03 01:36:57.024 | DEBUG    | lib.theme:get_theme:10 - Getting application theme from config file
2020-10-03 01:36:57.024 | DEBUG    | lib.config:get_key_value:21 - Attempting to read key 'theme' from the main config file C:\Users\ironb\AppData\Roaming\MSFS Mod Manager\config.ini
2020-10-03 01:36:57.025 | DEBUG    | lib.config:get_key_value:30 - Section key 'settings' found in config file
2020-10-03 01:36:57.025 | DEBUG    | lib.config:get_key_value:32 - Key 'theme' found in section
2020-10-03 01:36:57.025 | DEBUG    | lib.config:get_key_value:33 - Key 'theme' value: None
2020-10-03 01:36:57.025 | DEBUG    | lib.theme:get_theme:13 - FS theme is selected: False
2020-10-03 01:36:57.026 | DEBUG    | lib.theme:set_theme:19 - Writing theme selection to config file
2020-10-03 01:36:57.026 | DEBUG    | lib.config:set_key_value:46 - Attempting to write key 'theme' and value 'None' to the main config file C:\Users\ironb\AppData\Roaming\MSFS Mod Manager\config.ini
2020-10-03 01:36:57.027 | DEBUG    | lib.config:set_key_value:61 - Writing out config file
2020-10-03 01:36:57.035 | DEBUG    | lib.theme:set_theme:34 - Clearing application stylesheet
2020-10-03 01:36:57.035 | DEBUG    | lib.flight_sim:find_sim_folder:176 - Attempting to automatically locate simulator path
2020-10-03 01:36:57.035 | DEBUG    | lib.flight_sim:find_sim_folder:179 - Trying to find simulator path from config file
2020-10-03 01:36:57.035 | DEBUG    | lib.config:get_key_value:21 - Attempting to read key 'sim_folder' from the main config file C:\Users\ironb\AppData\Roaming\MSFS Mod Manager\config.ini
2020-10-03 01:36:57.036 | DEBUG    | lib.config:get_key_value:30 - Section key 'settings' found in config file
2020-10-03 01:36:57.036 | DEBUG    | lib.config:get_key_value:32 - Key 'sim_folder' found in section
2020-10-03 01:36:57.037 | DEBUG    | lib.config:get_key_value:33 - Key 'sim_folder' value: D:\D-Games\MSFlightSim2020
2020-10-03 01:36:57.037 | DEBUG    | lib.flight_sim:is_sim_packages_folder:161 - Testing if D:\D-Games\MSFlightSim2020 is MSFS sim packages folder
2020-10-03 01:36:57.037 | DEBUG    | lib.flight_sim:is_sim_packages_folder:165 - Folder D:\D-Games\MSFlightSim2020 is MSFS sim packages folder: True
2020-10-03 01:36:57.038 | DEBUG    | lib.flight_sim:find_sim_folder:182 - Config file sim path found and valid
2020-10-03 01:36:57.038 | DEBUG    | lib.version:check_version:92 - Checking if a new version is available
2020-10-03 01:36:57.038 | DEBUG    | lib.version:check_version_config:53 - Trying to read never version check from config file
2020-10-03 01:36:57.038 | DEBUG    | lib.config:get_key_value:21 - Attempting to read key 'never_version_check' from the main config file C:\Users\ironb\AppData\Roaming\MSFS Mod Manager\config.ini
2020-10-03 01:36:57.039 | DEBUG    | lib.config:get_key_value:30 - Section key 'settings' found in config file
2020-10-03 01:36:57.039 | DEBUG    | lib.config:get_key_value:36 - Unable to find key 'never_version_check' in config file
2020-10-03 01:36:57.039 | DEBUG    | lib.version:check_version_config:59 - Trying to read last version check from config file
2020-10-03 01:36:57.039 | DEBUG    | lib.config:get_key_value:21 - Attempting to read key 'last_version_check' from the main config file C:\Users\ironb\AppData\Roaming\MSFS Mod Manager\config.ini
2020-10-03 01:36:57.040 | DEBUG    | lib.config:get_key_value:30 - Section key 'settings' found in config file
2020-10-03 01:36:57.040 | DEBUG    | lib.config:get_key_value:32 - Key 'last_version_check' found in section
2020-10-03 01:36:57.040 | DEBUG    | lib.config:get_key_value:33 - Key 'last_version_check' value: 2020-10-03 00:27:29
2020-10-03 01:36:57.041 | DEBUG    | lib.version:check_version_config:65 - Trying to parse value 2020-10-03 00:27:29 to datetime
2020-10-03 01:36:57.043 | DEBUG    | lib.version:check_version_config:71 - Current time 2020-10-03 01:36:57.043750 is less than one day from last check 2020-10-03 00:27:29
2020-10-03 01:36:57.043 | DEBUG    | lib.flight_sim:get_enabled_mods:361 - Retrieving list of enabled mods


end:
020-10-03 01:36:57.461 | ERROR    | lib.flight_sim:parse_mod_manifest:337 - manifest.json could not be parsed
Traceback (most recent call last):

  File "main.py", line 40, in <module>

  File "widgets\main_widget.py", line 634, in refresh

  File "lib\flight_sim.py", line 368, in get_enabled_mods

> File "lib\flight_sim.py", line 335, in parse_mod_manifest

  File "json\__init__.py", line 296, in load

  File "c:\hostedtoolcache\windows\python\3.6.8\x64\lib\encodings\cp1252.py", line 23, in decode

UnicodeDecodeError: 'charmap' codec can't decode byte 0x81 in position 951: character maps to <undefined>

**Comment by feiercrack** [comment]
now in ur current version:

2020-10-06 11:52:35.219 | DEBUG    | lib.flight_sim:parse_mod_manifest:315 - Parsing manifest for D:\D-Games\MSFlightSim2020\Community\baddweapon-scenery-icelandoverhaul-v1-5
2020-10-06 11:52:35.220 | ERROR    | lib.flight_sim:parse_mod_manifest:328 - manifest.json could not be parsed
Traceback (most recent call last):

  File "main.py", line 40, in <module>

  File "widgets\main_widget.py", line 631, in refresh

  File "lib\flight_sim.py", line 359, in get_enabled_mods

> File "lib\flight_sim.py", line 326, in parse_mod_manifest

  File "json\__init__.py", line 296, in load

  File "c:\hostedtoolcache\windows\python\3.6.8\x64\lib\encodings\cp1252.py", line 23, in decode

UnicodeDecodeError: 'charmap' codec can't decode byte 0x81 in position 951: character maps to <undefined>
2020-10-06 11:52:35.222 | ERROR    | __main__:<module>:50 - Uncaught exception
Traceback (most recent call last):

  File "lib\flight_sim.py", line 326, in parse_mod_manifest

  File "json\__init__.py", line 296, in load

  File "c:\hostedtoolcache\windows\python\3.6.8\x64\lib\encodings\cp1252.py", line 23, in decode

UnicodeDecodeError: 'charmap' codec can't decode byte 0x81 in position 951: character maps to <undefined>


During handling of the above exception, another exception occurred:


Traceback (most recent call last):

> File "main.py", line 40, in <module>

  File "widgets\main_widget.py", line 631, in refresh

  File "lib\flight_sim.py", line 359, in get_enabled_mods

  File "lib\flight_sim.py", line 329, in parse_mod_manifest

lib.flight_sim.ManifestError: 'charmap' codec can't decode byte 0x81 in position 951: character maps to < undefined >

**Comment by NathanVaughn** [comment]
Thanks for the info, I believe I have fixed the issue. The `manifest.json` file in the Iceland overhaul mod had a funky text encoding that wasn't being handled properly. Additionally, I forgot to catch that error type, so there was no dialog being displayed and the program was immediately crashing.

Version [0.4.3](https://github.com/NathanVaughn/msfs-mod-manager/releases/tag/v0.4.3) should fix it. Let me know it works or doesn't

**Comment by feiercrack** [comment]
Awesome, great, works nearly perfect now!
2 Things to say:
1. Is it possible to choose the folder where all the "DISABLED" mods are going?
Cause i dont like to have them on my C: drive. Better would be my D. drive :-)
2. When i switch over to "FS-Theme" all letters went to "CAPITAL" letters..

But for now this is the best Manager i´ver ever tried (2 others are very buggy and not so comfortables as yours!
Thanks
cheers

**Comment by NathanVaughn** [comment]
1. That's a great idea, I added a new issue for it #35 
2. That's intentional, the MSFS UI is all uppercase, so I set that with [application stylesheet](https://github.com/NathanVaughn/msfs-mod-manager/blob/65a13cf8999d7add75c7268131a393bfd1388480/src/main/resources/base/fs_style.qss#L4). Now, why the table headers don't go back to lowercase when you disable the theme, until you restart the application? I'm not really sure, haven't been able to figure that out. It might be a Qt bug.
